drivers/bluetooth/hci/rpmsg.c

```diff
@@ -317,9 +325,42 @@ static int bt_rpmsg_open(void)
 	return 0;
 }
 
+static int bt_rpmsg_close(void)
+{
+	int err;
+
+	err = bt_hci_cmd_send_sync(BT_HCI_OP_RESET, NULL, NULL);
+	if (err) {
+		LOG_ERR("Sending reset command failed with: %d", err);
+		return err;
+	}
+
+	err = ipc_service_deregister_endpoint(&hci_ept);
+	if (err) {
+		LOG_ERR("Deregistering HCI endpoint failed with: %d", err);
+		return err;
+	}
+
+	const struct device *hci_ipc_instance =
+		DEVICE_DT_GET(DT_CHOSEN(zephyr_bt_hci_rpmsg_ipc));
+
+	err = ipc_service_close_instance(hci_ipc_instance);
+	if (err) {
+		LOG_ERR("Closing IPC service failed with: %d", err);
+		return err;
+	}
+
+#ifdef CONFIG_SOC_NRF5340_CPUAPP
+	cpunet_forceoff();
```

Comment:
Also use a weak function as I suggested for the setup and name the weak function, and name it `bt_hci_transport_teardown`.